driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -324,9 +324,14 @@ public ConnectionString(final String connectionString) {
                 userName = urldecode(userInfo);
             } else {
                 idx = userInfo.indexOf(":");
+                if (idx == 0) {
```

Comment:
Are these checks needed in the connection string - looking at the tests "should throw an exception if no username" it would have already failed when the MongoCredential was created?

Reply:
In this constructor, the MongoCredential is created at the end (line 392) after all checks have completed.

Reply:
If we remove these conditions, I think we'd get other exceptions thrown in the MongoCredential constructor.  But having the checks here allow the exception message to provide a little more context to the user.